R/Module-PatientLevelPredictionValidation.R

```diff
@@ -0,0 +1,254 @@
+PatientLevelPredictionValidationModule <- R6::R6Class(
+  classname = "PatientLevelPredictionValidationModule",
+  inherit = StrategusModule,
+  public = list(
+    #' @field tablePrefix The table prefix to append to the results tables
+    tablePrefix = "val_",
+    #' @description Initialize the module
+    initialize = function() {
+      super$initialize()
+    },
+    #' @description Executes the PatientLevelPrediction package to validate a
+    #' PLP model
+    #' @template connectionDetails
+    #' @template analysisSpecifications
+    #' @template executionSettings
+    execute = function(connectionDetails, analysisSpecifications, executionSettings) {
+      super$.validateCdmExecutionSettings(executionSettings)
+      super$execute(connectionDetails, analysisSpecifications, executionSettings)
+
+      jobContext <- private$jobContext
+      #cohortDefinitionSet <- super$.createCohortDefinitionSetFromJobContext()
+      workFolder <- jobContext$moduleExecutionSettings$workSubFolder
+      resultsFolder <- jobContext$moduleExecutionSettings$resultsSubFolder
+
+      #library(PatientLevelPrediction)
+      private$.message("Validating inputs")
+      inherits(jobContext, 'list')
+
+      if (is.null(jobContext$settings)) {
+        stop("Analysis settings not found in job context")
+      }
+      # if (is.null(jobContext$sharedResources)) {
+      #   stop("Shared resources not found in job context")
+      # }
+      if (is.null(jobContext$moduleExecutionSettings)) {
+        stop("Execution settings not found in job context")
+      }
+
+      #workFolder <- jobContext$moduleExecutionSettings$workSubFolder
+      #resultsFolder <- jobContext$moduleExecutionSettings$resultsSubFolder
+
+      private$.message("Executing PLP Validation")
+      #moduleInfo <- getModuleInfo()
+
+      # find where cohortDefinitions are as sharedResources is a list
+      # cohortDefinitionSet <- createCohortDefinitionSetFromJobContext(
+      #   sharedResources = jobContext$sharedResources,
+      #   settings = jobContext$settings
+      # )
+
+      # check the model locations are valid and apply model
+      upperWorkDir <- dirname(jobContext$moduleExecutionSettings$workFolder) # AGS: NOTE - Using the "root" folder as the expection is that the ModelTransferModule output is here
+      modelTransferFolder <- sort(dir(upperWorkDir, pattern = 'ModelTransferModule'), decreasing = T)[1]
+
+      modelSaveLocation <- file.path( upperWorkDir, modelTransferFolder, 'models') # hack to use work folder for model transfer
+      modelInfo <- private$.getModelInfo(modelSaveLocation)
+
+      designs <- list()
+      for (i in seq_len(nrow(modelInfo))) {
+        df <- modelInfo[i, ]
+
+        design <- PatientLevelPrediction::createValidationDesign(
+          targetId = df$target_id[1],
+          outcomeId = df$outcome_id[1],
+          plpModelList = as.list(df$modelPath),
+          restrictPlpDataSettings = jobContext$settings[[1]]$restrictPlpDataSettings,
```

Comment:
Leaving this here for @egillax - just wanted to note that the module's settings implies a list of lists for the `validationComponentsList` but here we're only using the first element of the list?